rio_stac/stac.py

```diff
@@ -0,0 +1,135 @@
+"""Create STAC Item from a rasterio dataset."""
+
+import datetime as pydatetime
+import os
+from contextlib import ExitStack
+from typing import Any, Dict, List, Optional, Sequence, Union
+
+import pystac
+import rasterio
+from rasterio import warp
+from rasterio.io import DatasetReader, DatasetWriter, MemoryFile
+from rasterio.vrt import WarpedVRT
+
+
+def bbox_to_geom(bbox: Sequence[float]) -> Dict:
+    """Return a geojson geometry from a bbox."""
+    # TODO: Handle dateline geometry
+    return {
+        "type": "Polygon",
+        "coordinates": [
+            [
+                [bbox[0], bbox[3]],
+                [bbox[0], bbox[1]],
+                [bbox[2], bbox[1]],
+                [bbox[2], bbox[3]],
+                [bbox[0], bbox[3]],
+            ]
+        ],
+    }
+
+
+def get_metadata(
+    source: Union[str, DatasetReader, DatasetWriter, WarpedVRT, MemoryFile]
```

Comment:
Is it necessary to allow `source` to include `DatasetWriter`?

Reply:
why not? 

In theory you could create a STAC item while you are creating your raster

Reply:
Yeah that's a good point; I hadn't thought of that.